src/pudl/analysis/timeseries_cleaning.py

```diff
@@ -1649,10 +1656,11 @@ def _add_simulated_flag_col(
         left_on=["id_col", "period"],
         right_on=["reference_id_col", "period"],
         how="inner",
+        validate="many_to_one",
     )
 
     # Filter to hours where values were flagged (NULL) in reference month
-    flagged = simulation_df[simulation_df["value_col"].isnull()]
+    flagged = simulation_df[simulation_df["flags"].notnull()]
```

Comment:
Switched to checking for flags directly, but doesn't actually change the results.